core/templates/dev/head/pages/admin/roles_tab/RoleGraphDirective.js

```diff
@@ -0,0 +1,222 @@
+
+            for (var i = 0; i < $scope.augmentedLinks.length; i++) {
+              var link = $scope.augmentedLinks[i];
+              if (link.source.label !== link.target.label) {
+                var sourcex = link.source.xLabel;
+                var sourcey = link.source.yLabel;
+                var targetx = link.target.xLabel;
+                var targety = link.target.yLabel;
+
+                if (sourcex === targetx && sourcey === targety) {
+                  // TODO(sll): Investigate why this happens.
+                  return;
+                }
+
+                var sourceWidth = link.source.width;
+                var sourceHeight = link.source.height;
+                var targetWidth = link.target.width;
+                var targetHeight = link.target.height;
+
+                var dx = targetx - sourcex;
+                var dy = targety - sourcey;
+
+                /* Fractional amount of truncation to be applied to the end of
+                   each link. */
+                var startCutoff = (sourceWidth / 2) / Math.abs(dx);
+                var endCutoff = (targetWidth / 2) / Math.abs(dx);
+                if (dx === 0 || dy !== 0) {
+                  startCutoff = (
+                    (dx === 0) ? (sourceHeight / 2) / Math.abs(dy) :
+                    Math.min(startCutoff, (sourceHeight / 2) / Math.abs(dy)));
+                  endCutoff = (
+                    (dx === 0) ? (targetHeight / 2) / Math.abs(dy) :
+                    Math.min(endCutoff, (targetHeight / 2) / Math.abs(dy)));
+                }
+
+                var dxperp = targety - sourcey;
+                var dyperp = sourcex - targetx;
+                var norm = Math.sqrt(dxperp * dxperp + dyperp * dyperp);
+                dxperp /= norm;
+                dyperp /= norm;
+
+                var midx = sourcex + dx / 2 + dxperp * (sourceHeight / 4);
+                var midy = sourcey + dy / 2 + dyperp * (targetHeight / 4);
+                var startx = sourcex + startCutoff * dx;
+                var starty = sourcey + startCutoff * dy;
+                var endx = targetx - endCutoff * dx;
+                var endy = targety - endCutoff * dy;
+
+                // Draw a quadratic bezier curve.
+                $scope.augmentedLinks[i].d = (
+                  'M' + startx + ' ' + starty + ' Q ' + midx + ' ' + midy +
+                  ' ' + endx + ' ' + endy);
+
+                // Style links if link properties and style mappings are
+                // provided
+                if (links[i].hasOwnProperty('linkProperty') &&
+                    $scope.linkPropertyMapping) {
+                  if ($scope.linkPropertyMapping.hasOwnProperty(
+                        links[i].linkProperty)) {
+                    $scope.augmentedLinks[i].style = (
+                      $scope.linkPropertyMapping[links[i].linkProperty]);
+                  }
+                }
+              }
+            }
+
+            var nodeStrokeWidth = '1';
```

Comment:
these seem kind of magical constants. Since we only have 1 type of node/link, we can probably just put the styles in the CSS.